nifi-nar-bundles/nifi-ccda-bundle/nifi-ccda-processors/src/main/java/org/apache/nifi/processors/ccda/ExtractCCDAAttributes.java

```diff
@@ -0,0 +1,383 @@
+        String name = element.getClass().getName();
+        Map<String, String> jexlMap = (HashMap<String, String>) processMap.get(name); // get JEXL mappings for this element
+        getLogger().debug("Processing " + name);
+
+        if (jexlMap == null) {
+            getLogger().info("Missing mapping for element " + name);
+            return null;
+        }
+        for (Entry<String, String> entry : jexlMap.entrySet()) { // evaluate JEXL for each child element
+            jexlCtx.set("element", element);
+            JexlExpression jexlExpr = jexl.createExpression(entry.getValue());
+            Object value = jexlExpr.evaluate(jexlCtx);
+            String key = entry.getKey();
+            String prefix = parent != null ? parent + "." + key : key;
+            addElement(map, prefix, key, value);
+        }
+        timingStats.add(String.format("Processed %s in %d ms", name, System.currentTimeMillis() - start));
+        return map;
+    }
+
+    /**
+     * Adds element to the JSON map and attribute list based on the type
+     * @param map       JSON map
+     * @param prefix    parent key as prefix
+     * @param key       element key
+     * @param value     element value
+     */
+    protected void addElement(Map<String, Object> map, String prefix, String key, Object value) {
+        // if the value is a String, add it to final attribute list
+        // else process it further until we have a String representation
+        if (value instanceof String) {
+            if(value != null && !((String) value).isEmpty()) {
+                map.put(key, value);
+                attributes.put(prefix, (String) value);
+            }
+        } else if (value instanceof List) {
+            if(value != null && !((List) value).isEmpty()) {
+                map.put(key, processList(prefix, (List) value));
+            }
+        } else if (value != null) { // process element further
+            map.put(key, processElement(prefix, value));
+        }
+    }
+
+    /**
+     * Iterate through the list and calls processElement to process each element
+     * @param key       key used while calling processElement
+     * @param value     value is the individual Object being processed
+     * @return          list of elements
+     */
+    protected List<Object> processList(String key, List value) {
+        List<Object> items = new ArrayList<Object>();
+        String keyFormat = value.size() > 1 ? "%s_%02d" : "%s";
+        for (Object item : value) { // iterate over all elements and process each element
+            items.add(processElement(String.format(keyFormat, key, items.size() + 1), item));
+        }
+        return items;
+    }
+
+    protected ClinicalDocument loadDocument(InputStream inputStream, Boolean skipValidation) {
+        long start = System.currentTimeMillis();
+        ClinicalDocument cd = null;
+
+        try {
+            getLogger().info("Loading document");

```

Comment:
This should be a debug() or trace() since it's going to get called for every onTrigger().